packages/openspp/src/Adaptor.js

```diff
@@ -0,0 +1,261 @@
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function getHousehold(householdName, callback=false) {
+  if (sppConnector === null) {
+    login(state);
+  }
+  return state => {
+    let defaultDomain = [
+      ["is_registrant", "=", true],
+      ["is_group", "=", true],
+      ["name", "=", householdName]
+    ];
+    let defaultOrder = "id desc";
+    let defaultFields = [];
+    let options = {
+      domain: defaultDomain,
+      limit: 1,
+      order: defaultOrder,
+      fields: defaultFields
+    };
+    sppConnector.search_read("res.partner", options, (err, household) => {
+      if (err) {
+        Log.error(err);
+      }
+      if (!household) {
+        return Log.warn(`Household ${householdName} not found!`);
+      }
+      Log.info(`Household ${householdName} found!`);
+      let nextState = composeNextState(state, household);
+      if (callback) {
+        return callback(nextState);
+      }
+      return nextState;
+    });
+  };
+};
+
+/**
+ * get household information from OpenSPP
+ * @public
+ * @example
+ * getHouseholdById(641)
+ * @function
+ * @param {string} householdId - The id of the household
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function getHouseholdById(householdId, callback=false) {
+  if (sppConnector === null) {
+    login(state);
+  }
+  return state => {
+    let defaultDomain = [
+      ["is_registrant", "=", true],
+      ["is_group", "=", true],
+      ["id", "=", householdId]
+    ];
+    let defaultOrder = "id desc";
+    let defaultFields = [];
+    let options = {
+      domain: defaultDomain,
+      limit: 1,
+      order: defaultOrder,
+      fields: defaultFields
+    };
+    sppConnector.search_read("res.partner", options, (err, household) => {
+      if (err) {
+        Log.error(err);
+      }
+      if (!household) {
+        return Log.warn(`Household with id=${householdId} not found!`);
+      }
+      Log.info(`Household with id=${householdId} found!`);
+      let nextState = composeNextState(state, household);
+      if (callback) {
+        return callback(nextState);
+      }
+      return nextState;
+    });
+  };
+};
+
+/**
+ * get household members information from OpenSPP
+ * @public
+ * @example
+ * getHouseholdMembers("6410000117")
+ * @function
+ * @param {string} householdName - The name of the household
+ * @param {number} [offset=0] - Offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function getHouseholdMembers(householdName, offset=0, callback=false) {
+  if (sppConnector === null) {
+    login(state);
+  }
+  return state => {
+    let defaultDomain = [
+      ["is_ended", "=", false],
+      ["group.name", "=", householdName]
+    ];
+    let defaultFields = [
+      "individual", "kind", "start_date", "ended_date",
+      "individual_birthdate", "individual_gender"
```

Comment:
Those field ("individual_birthdate", "individual_gender") are not in the default implementation of OpenSPP I think.
Try the code with a fresh install with openspp-docker.

Reply:
You are right, this is not in OpenSPP, but is implemented by you in OpenG2P repo. [(link here)](https://github.com/OpenG2P/openg2p-registry/blob/c3ff14407de55c51e64f7dc598b736a350b26e7d/g2p_registry_membership/models/group_membership.py#L31)
And I think that it is good to have.